cou/steps/nova.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Canonical Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Functions for prereq steps relating to nova."""
+import logging
+
+from cou.exceptions import UnitNotFound
+from cou.utils.juju_utils import Model
+
+logger = logging.getLogger(__name__)
+
+
+async def archive(model: Model, *, batch_size: int) -> None:
+    """Archive data on a nova-cloud-controller unit.
+
+    The archive-data action only runs a single batch,
+    so we run it in a loop until completion.
+    See also:
+    https://charmhub.io/nova-cloud-controller/actions#archive-data
+    https://docs.openstack.org/project-deploy-guide/charm-deployment-guide/wallaby/upgrade-openstack.html#archive-old-database-data
+    :param model: juju model to work with
+    :type model: Model
+    :param batch_size: batch-size to pass to the archive-data action
+        (default is 1000; decrease if performance issues seen)
+    :type batch_size: int
+    """  # noqa: E501 line too long
+    unit_name: str = await _get_nova_cloud_controller_unit_name(model)
+    while True:
```

Comment:
**nitpick:** 

I think you miss the error handling here. What if the action is failed?

Reply:
I do not agree, since if action failed cou will not catch COUException but some exception from Juju -> user will see tracback and message from libjuju, what is not what we want.